Author: Judit Foglszinger <fgrfgr@freenet.de>
Description: Fixing typo in output for x-icon (Closes: #593488).

diff -Naurp file.orig/magic/Magdir/msdos file/magic/Magdir/msdos
--- file.orig/magic/Magdir/msdos	2010-09-18 11:42:55.297853937 +0200
+++ file/magic/Magdir/msdos	2010-09-18 11:43:05.326837856 +0200
@@ -484,7 +484,7 @@
 
 # Windows icons (Ian Springer <ips@fpk.hp.com>)
 0	string	\000\000\001\000	MS Windows icon resource
-!:mime	image/x-ico
+!:mime	image/x-icon
 >4	byte	1			- 1 icon
 >4	byte	>1			- %d icons
 >>6	byte	>0			\b, %dx
