Subject: Pyzip improvements (FC Stegerman)
Origin: FILE5_44-9-gdc71304b <https://github.com/file/file/commit/FILE5_44-9-gdc71304b>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Sat Dec 31 20:24:08 2022 +0000

--- a/magic/Magdir/archive
+++ b/magic/Magdir/archive
@@ -1876,9 +1876,14 @@
 # https://en.wikipedia.org/wiki/ZIP_(file_format)#End_of_central_directory_record_(EOCD)
 # by Michal Gorny <mgorny@gentoo.org>
 -2	uleshort	0
->&-22	string	PK\005\006	Zip archive, with extra data prepended
+>&-22	string	PK\005\006
+# without #!
+>>0	string	!#!	Zip archive, with extra data prepended
 !:mime	application/zip
 !:ext zip/cbz
+# with #!
+>>0	string/w	#!\ 	a
+>>>&-1	string/T	x	%s script executable (Zip archive)
 
 # ACE archive (from http://www.wotsit.org/download.asp?f=ace)
 # by Stefan `Sec` Zehl <sec@42.org>
