Subject: Detect .smali files (FC Stegerman)
Origin: FILE5_44-17-g4c8ee2e7 <https://github.com/file/file/commit/FILE5_44-17-g4c8ee2e7>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Thu Jan 12 00:09:46 2023 +0000

--- a/magic/Magdir/android
+++ b/magic/Magdir/android
@@ -207,3 +207,8 @@
 >8	string	>000	dex section version: %s,
 >12	lelong	>0	number of dex files: %d,
 >16	lelong	>0	verifier deps size: %d
+
+# Disassembled DEX files
+0	string/t	.class\x20
+>&0	regex/512	\^\\.super\x20L.*;$	disassembled Android DEX Java class (smali/baksmali)
+!:ext	smali
