Subject: Reduce the number of X's to appease linitian (Christoph Biedl)
Origin: FILE5_28-6-g0971151
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Mon Jun 27 19:07:20 2016 +0000

--- a/doc/file.man
+++ b/doc/file.man
@@ -312,7 +312,7 @@
 never read them.
 .It Fl P , Fl Fl parameter Ar name=value
 Set various parameter limits.
-.Bl -column "elf_phnum" "Default" "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX" -offset indent
+.Bl -column "elf_phnum" "Default" "XXXXXXXXXXXXXXXXXXXXXXXXXXX" -offset indent
 .It Sy "Name" Ta Sy "Default" Ta Sy "Explanation"
 .It Li indir Ta 15 Ta recursion limit for indirect magic
 .It Li name Ta 30 Ta use count limit for name/use magic
