Subject: Fix documentation to match the source
Origin: FILE5_35-42-ge6bf546d <https://github.com/file/file/commit/FILE5_35-42-ge6bf546d>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Mon Jan 21 14:56:53 2019 +0000

--- a/doc/magic.man
+++ b/doc/magic.man
@@ -116,13 +116,13 @@
 .It B
 A byte length (default).
 .It H
-A 4 byte big endian length.
+A 2 byte big endian length.
 .It h
 A 2 byte little endian length.
 .It L
 A 4 byte big endian length.
 .It l
-A 2 byte little endian length.
+A 4 byte little endian length.
 .It J
 The length includes itself in its count.
 .El
