Subject: PR/19: Recognize ia64 and amd64 COFF files
Origin: FILE5_34-26-gc673ce51 <https://github.com/file/file/commit/FILE5_34-26-gc673ce51>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Wed Aug 1 10:34:03 2018 +0000
Bug-Debian: https://bugs.debian.org/697846

--- a/magic/Magdir/coff
+++ b/magic/Magdir/coff
@@ -26,6 +26,10 @@
 >>0	uleshort	0x0550		Hitachi SH little-endian
 # executable (RISC System/6000 V3.1) or obj module (./ibm6000)
 #>>0	uleshort	0x01DF
+# MS Windows COFF Intel Itanium, AMD64
+# https://msdn.microsoft.com/en-us/library/windows/desktop/ms680313(v=vs.85).aspx
+>>0	uleshort	0x0200		Intel ia64
+>>0	uleshort	0x8664		Intel amd64
 # TODO for other COFFs
 #>>0	uleshort	0xABCD		COFF_TEMPLATE
 >>0	default		x
--- a/magic/Magdir/intel
+++ b/magic/Magdir/intel
@@ -1,6 +1,6 @@
 
 #------------------------------------------------------------------------------
-# $File: intel,v 1.16 2017/11/14 15:48:36 christos Exp $
+# $File: intel,v 1.17 2018/08/01 10:34:03 christos Exp $
 # intel:  file(1) magic for x86 Unix
 #
 # Various flavors of x86 UNIX executable/object (other than Xenix, which
@@ -42,6 +42,10 @@
 #>12	lelong		>0		not stripped
 # no hint found, that at offset 22 is version
 #>22	leshort		>0		- version %d
+0	leshort		0x0200
+>0	use				display-coff
+0	leshort		0x8664
+>0	use				display-coff
 
 # rom: file(1) magic for BIOS ROM Extensions found in intel machines
 #      mapped into memory between 0xC0000 and 0xFFFFF
