Subject: Fix cut-n-pasto for regex_max (Vsevolod Stakhov)
Origin: FILE5_35-25-g48052fcf <https://github.com/file/file/commit/FILE5_35-25-g48052fcf>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Tue Dec 11 14:41:11 2018 +0000

--- a/src/magic.c
+++ b/src/magic.c
@@ -604,7 +604,7 @@
 		ms->elf_notes_max = (uint16_t)*(const size_t *)val;
 		return 0;
 	case MAGIC_PARAM_REGEX_MAX:
-		ms->elf_notes_max = (uint16_t)*(const size_t *)val;
+		ms->elf_regex_max = (uint16_t)*(const size_t *)val;
 		return 0;
 	case MAGIC_PARAM_BYTES_MAX:
 		ms->bytes_max = *(const size_t *)val;
