Subject: PR/316: karagian: Bump strength to beat amd 29k a.out
Origin: FILE5_41-76-g394df3ed <https://github.com/file/file/commit/FILE5_41-76-g394df3ed>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Sat Feb 19 22:35:32 2022 +0000

--- a/magic/Magdir/database
+++ b/magic/Magdir/database
@@ -151,6 +151,7 @@
 # https://www.clicketyclick.dk/databases/xbase/format/dbf.html
 # inspect VVYYMMDD , where 1<= MM <= 12 and 1<= DD <= 31
 0	ubelong&0x0000FFFF		<0x00000C20
+!:strength +10
 # skip Infocom game Z-machine
 >2		ubyte			>0
 # skip Androids *.xml
