Subject: PR/10: Add sketchup
Origin: FILE5_34-29-g4efd402b <https://github.com/file/file/commit/FILE5_34-29-g4efd402b>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Thu Aug 2 06:23:27 2018 +0000
Bug-Debian: https://bugs.debian.org/903693

--- a/magic/Magdir/cad
+++ b/magic/Magdir/cad
@@ -183,3 +183,8 @@
 >>>>&1	search/8192	AC1018			\b, version 2004
 >>>>&1	search/8192	AC1021			\b, version 2007
 >>>>&1	search/8192	AC1024			\b, version 2010
+
+# The Sketchup 3D model format https://www.sketchup.com/
+0	string	\xff\xfe\xff\x0e\x53\x00\x6b\x00\x65\x00\x74\x00\x63\x00\x68\x00\x55\x00\x70\x00\x20\x00\x4d\x00\x6f\x00\x64\x00\x65\x00\x6c\x00	SketchUp Model
+!:mime application/vnd.sketchup.skp
+!:ext skp
