Subject: bump more
Origin: FILE5_25-5-gba7b688
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Wed Sep 16 22:19:54 2015 +0000

--- a/magic/Magdir/python
+++ b/magic/Magdir/python
@@ -26,16 +26,16 @@
 0	belong		0xee0c0d0a	python 3.4 byte-compiled
 
 0	search/1/w	#!\ /usr/bin/python	Python script text executable
-!:strength + 10
+!:strength + 15
 !:mime text/x-python
 0	search/1/w	#!\ /usr/local/bin/python	Python script text executable
-!:strength + 10
+!:strength + 15
 !:mime text/x-python
 0	search/1	#!/usr/bin/env\ python	Python script text executable
-!:strength + 10
+!:strength + 15
 !:mime text/x-python
 0	search/10	#!\ /usr/bin/env\ python	Python script text executable
-!:strength + 10
+!:strength + 15
 !:mime text/x-python
 
 
