Subject: Fix escaping of periods
Origin: FILE5_38-99-g2f3e66e1 <https://github.com/file/file/commit/FILE5_38-99-g2f3e66e1>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Mon Apr 27 01:50:36 2020 +0000
Bug-Debian: https://bugs.debian.org/949878

--- a/magic/Magdir/rst
+++ b/magic/Magdir/rst
@@ -7,5 +7,5 @@
 >&0	regex/256	\^[\=]+$
 >>&0	search/512	:Author:	ReStructuredText file
 >>&0	default		x
->>>&0	regex/512	\^\.\.[A-Za-z]	ReStructuredText file
+>>>&0	regex/512	\^\\.\\.[A-Za-z]	ReStructuredText file
 !:ext	rst
