Subject: Change default for name/use to 50
Origin: Part of FILE5_38-65-gdf476c81 <https://github.com/file/file/commit/FILE5_38-65-gdf476c81>
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Thu Mar 19 20:41:11 2020 +0000
Bug-Debian: https://bugs.debian.org/928009

--- a/src/file.h
+++ b/src/file.h
@@ -437,7 +437,7 @@
 	uint16_t regex_max;
 	size_t bytes_max;		/* number of bytes to read from file */
 #define	FILE_INDIR_MAX			50
-#define	FILE_NAME_MAX			30
+#define	FILE_NAME_MAX			50
 #define	FILE_ELF_SHNUM_MAX		32768
 #define	FILE_ELF_PHNUM_MAX		2048
 #define	FILE_ELF_NOTES_MAX		256
