Subject: Extra magic file, to be upstreamed
Author: Christoph Biedl <debian.axhn@manchmal.in-ulm.de>
Date: 2018-07-18
Forwarded: not-yet

--- a/magic/Makefile.am
+++ b/magic/Makefile.am
@@ -10,6 +10,7 @@
 MAGIC_FRAGMENTS = \
 $(MAGIC_DIR)/Header \
 $(MAGIC_DIR)/Localstuff \
+$(MAGIC_FRAGMENT_DIR)/../../debian/debian-extra-magic \
 $(MAGIC_FRAGMENT_DIR)/acorn \
 $(MAGIC_FRAGMENT_DIR)/adi \
 $(MAGIC_FRAGMENT_DIR)/adventure \
