Subject: Clarify manpage: seccomp is disabled for the time being
Author: Christoph Biedl <debian.axhn@manchmal.in-ulm.de>
Date: 2019-01-11
Bug-Debian: https://bugs.debian.org/917930

--- a/doc/file.man
+++ b/doc/file.man
@@ -387,6 +387,10 @@
 .Fl z
 option is specified and the built-in decompressors are not available.
 On systems where sandboxing is not available, this option has no effect.
+.Pp
+.Em Note:
+This Debian version of file was built without seccomp support, so this option
+has no effect.
 .It Fl v , Fl Fl version
 Print the version of the program and exit.
 .It Fl z , Fl Fl uncompress
