Subject: Revert: PR/591: Print if the file has debug info
Origin: FILE5_29-39-g0e78a704
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Sun Jan 29 19:34:24 2017 +0000

--- a/src/readelf.c
+++ b/src/readelf.c
@@ -1185,7 +1185,7 @@
 {
 	Elf32_Shdr sh32;
 	Elf64_Shdr sh64;
-	int stripped = 1, has_debug_info = 1;
+	int stripped = 1;
 	size_t nbadcap = 0;
 	void *nbuf;
 	off_t noff, coff, name_off;
@@ -1216,10 +1216,8 @@
 			return -1;
 		}
 		name[namesize] = '\0';
-		if (strcmp(name, ".debug_info") == 0) {
-			has_debug_info = 1;
+		if (strcmp(name, ".debug_info") == 0)
 			stripped = 0;
-		}
 
 		if (pread(fd, xsh_addr, xsh_sizeof, off) < (ssize_t)xsh_sizeof) {
 			file_badread(ms);
@@ -1375,10 +1373,6 @@
 
 	if (file_printf(ms, ", %sstripped", stripped ? "" : "not ") == -1)
 		return -1;
-	if (has_debug_info) {
-		if (file_printf(ms, ", with debug_info") == -1)
-			return -1;
-	}
 	if (cap_hw1) {
 		const cap_desc_t *cdp;
 		switch (mach) {
