Subject: Revert: Add IPS magic (Thomas Klausner)
Origin: FILE5_29-46-g53b8d4b0
Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Wed Feb 8 21:53:06 2017 +0000

--- a/magic/Magdir/console
+++ b/magic/Magdir/console
@@ -515,10 +515,6 @@
 >3	byte		x	version %d,
 >4	byte		x	%d tracks
 
-# IPS Patch Files from: From: Thomas Klausner <tk@giga.or.at>
-# see http://zerosoft.zophar.net/ips.php
-0	string	PATCH			IPS patch file
-
 # Playstations Patch Files from: From: Thomas Klausner <tk@giga.or.at>
 0	string	PPF30			Playstation Patch File version 3.0
 >5	byte	0			\b, PPF 1.0 patch
