Subject: Detect smali files
Author: Christoph Biedl <debian.axhn@manchmal.in-ulm.de>
Date: 2022-12-30
Forwarded: https://mailman.astron.com/pipermail/file/2022-December/001049.html

--- a/magic/Magdir/android
+++ b/magic/Magdir/android
@@ -207,3 +207,8 @@
 >8	string	>000	dex section version: %s,
 >12	lelong	>0	number of dex files: %d,
 >16	lelong	>0	verifier deps size: %d
+
+# Disassembled DEX files
+0	string/t	.class\x20
+>&0	regex/512	\^\\.super\x20L.*;$	disassembled Android DEX Java class (smali/baksmali)
+!:ext	smali
